R/utils.R

```diff
@@ -20,39 +20,19 @@ formatSolveBioResponse <- function (res, raw = FALSE) {
 
 }
 
-formatSolveBioQueryResponse <- function (id, res, raw = FALSE, row.names = NULL) {
+formatSolveBioQueryResponse <- function (res, raw = FALSE, col.names = NULL) {
     # res will be the output of formatSolveBioResponse
-    if (!raw & is.data.frame(res$results)) {
-        # Append column "_id" if it's not already there
-        if (!"_id" %in% row.names) {
-            row.names <- c(row.names, "_id")
+    if (!raw & is.data.frame(res$results) & !is.null(col.names)) {
+        # Flatten the data frame
+        res$results <- as.data.frame(res$results, col.names = col.names$name)
```

Comment:
If title is used, how does $name work?

Reply:
This part of the code is practically used only when querying Objects (files) and the names for the columns are actually the ones written in the file.